File: Capítulo_14/Algoritmos gulosos/Problema do número mínimo de moedas para troco/Código em Python/minCoinChange.py
# Algoritmo guloso min_coin_change
def min_coin_change(coins, amount):
    change = []
    total = 0
    for i in range(len(coins) - 1, -1, -1):
        coin = coins[i]

        while total + coin <= amount:
            change.append(coin)
            total += coin

    return change

# Testando algoritmo guloso minCoinChange
coins = [1, 5, 10, 25]
amount = 36
print(min_coin_change(coins, amount))

File: Capítulo_14/Algoritmos gulosos/Problema do número mínimo de moedas para troco/minCoinChange.js
// Algoritmo guloso minCoinChange
function minCoinChange(coins, amount) {
    const change = [];
    let total = 0;
    for (let i = coins.length; i >= 0; i--) {
        const coin = coins[i];
        while (total + coin <= amount) {
            change.push(coin);
            total += coin;
        }
    }
    return change;
}

// Testando algoritmo guloso minCoinChange
let coins = [1, 5, 10, 25];
let amount = 36;
console.log(minCoinChange(coins, amount));
